dpm/client/__init__.py

```diff
@@ -149,6 +154,43 @@ def publish(self, publisher=None):
         # Return published datapackage url
         return self.server + '/%s/%s' % (self.username, self.datapackage.descriptor['name'])
 
+    def _prepare_files_authorize(self):
+        files = filter(isfile, listdir(self.datapackage.base_path))
+
+        accepted_descriptor = 'datapackage.json'
```

Comment:
Why do we need to check through file list for `datapackage.json` - we know it must be there as this is a data package (if it is not there we should even attempt the publish ...)